### ISSUE TITLE
The only thing vi-mode does is ignore some key presses

### ISSUE THREAD

**Issue by SaffronSnail** [robbyrussell/oh-my-zsh#5746]
I used to use `bindkey -v` to use vi-like keybindings in zsh. After installing oh-my-zsh, this stopped working. I changed the line to `plugins=(vi-mode)`. Now if I hit escape and try to hit command keys it will ignore some number of key presses and then start inserting again. From what I can tell, it will skip one key press if I start by hitting '$' and two key presses if I start by hitting '0'. Here is a mapping from key sequences to observed behavior:

```
ESC 0 0 0 => one 0 is inserted
ESC 0 0 $ => one $ is inserted
ESC 0 $ $ => one $ is inserted
ESC $ $ $ => two $'s are inserted
ESC $ $ 0 => one $ followed by one 0 is inserted
ESC $ 0 0 => two 0's are inserted
ESC $ 0 $ => one 0 followed by one $ is inserted
ESC 0 $ 0 => one 0 is inserted
```
I am running Arch Linux with zsh verison 5.3.1. I have no other oh-my-zsh plugins installed. I have zsh-syntax-highlighting installed, but disabling it did not change the behavior of the program.

**Comment by slavaGanzin** [comment]
Can't reproduce on 
zsh 5.3.1
archlinux
urxvt

Can you try to use another terminal, like urxvt?

**Comment by slavaGanzin** [comment]
I found behavior you talking about.
I cut all set -o vi, binkey -v and vi-mode plugin from my zsh.

"ESC 0 0 0 => one 0 is inserted" is emacs-mode.

So you should overlook your zshrc and all includes.

**Comment by SaffronSnail** [comment]
Alright, I found two problems with my zshrc, and after resolving them everything appears to be working as expected.

The first problem that I had was that I thought that the line `plugins=(vi-mode)` would enable the plugin, not just load it, so I had removed my `bindkey -v` line when I discovered that this plugin exists.

The second problem that I had was that I sourced oh-my-zsh at the end of my zshrc, after `plugins=(vi-mode)`. I did this because I also use zsh-syntax-highlighting which says that it needs to be the last thing sourced, and it seemed natural to group the sourcing of external files.

Thanks for the help, much appreciated.
